flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -0,0 +1,344 @@
+	 * @throws Exception
+	 */
+	public static ActorSystem startActorSystem(Configuration configuration,
+										String listeningAddress,
+										String portRangeDefinition,
+										Logger logger) throws Exception {
+
+		// parse port range definition and create port iterator
+		Iterator<Integer> portsIterator;
+		try {
+			portsIterator = NetUtils.getPortRangeFromString(portRangeDefinition);
+		} catch (Exception e) {
+			throw new IllegalArgumentException("Invalid port range definition: " + portRangeDefinition);
+		}
+
+		while (portsIterator.hasNext()) {
+			// first, we check if the port is available by opening a socket
+			// if the actor system fails to start on the port, we try further
+			ServerSocket availableSocket = NetUtils.createSocketFromPorts(
+				portsIterator,
+				new NetUtils.SocketFactory() {
+					@Override
+					public ServerSocket createSocket(int port) throws IOException {
+						return new ServerSocket(port);
+					}
+				});
+
+			int port;
+			if (availableSocket == null) {
+				throw new BindException("Unable to allocate further port in port range: " + portRangeDefinition);
+			} else {
+				port = availableSocket.getLocalPort();
+				try {
+					availableSocket.close();
+				} catch (IOException ignored) {}
+			}
+
+			try {
+				return startActorSystem(configuration, listeningAddress, port, logger);
+			}
+			catch (Exception e) {
+				// we can continue to try if this contains a netty channel exception
+				Throwable cause = e.getCause();
+				if (!(cause instanceof org.jboss.netty.channel.ChannelException ||
+						cause instanceof java.net.BindException)) {
+					throw e;
+				} // else fall through the loop and try the next port
+			}
+		}
+
+		// if we come here, we have exhausted the port range
+		throw new BindException("Could not start actor system on any port in port range "
+			+ portRangeDefinition);
+	}
+
+	/**
+	 * Starts an Actor System at a specific port.
+	 * @param configuration The Flink configuration.
+	 * @param listeningAddress The address to listen at.
+	 * @param listeningPort The port to listen at.
+	 * @param logger the logger to output log information.
+	 * @return The ActorSystem which has been started.
+	 * @throws Exception
+	 */
+	public static ActorSystem startActorSystem(
+						Configuration configuration,
+						String listeningAddress,
+						int listeningPort,
+						Logger logger) throws Exception {
+
+		String hostPortUrl = NetUtils.hostAndPortToUrlString(listeningAddress, listeningPort);
+		logger.info("Trying to start actor system at " + hostPortUrl);
+
+		try {
+			Config akkaConfig = AkkaUtils.getAkkaConfig(
+				configuration,
+				new scala.Some<>(new scala.Tuple2<String, Object>(listeningAddress, listeningPort))
+			);
+			if (logger.isDebugEnabled()) {
+				logger.debug("Using akka configuration\n " + akkaConfig);
+			}
+
+			ActorSystem actorSystem = AkkaUtils.createActorSystem(akkaConfig);
+			logger.info("Actor system started at " + hostPortUrl);
```

Comment:
we usually use info("some log string {}", value); to avoid string concat when logging is disabled.